src/ucp/rndv/rndv_rkey_ptr.c

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "proto_rndv.inl"
+
+
+enum {
+    UCP_PROTO_RNDV_RKEY_PTR_STAGE_FETCH = UCP_PROTO_STAGE_START,
+    UCP_PROTO_RNDV_RKEY_PTR_STAGE_ATS
+};
+
+static ucs_status_t
+ucp_proto_rndv_rkey_ptr_init(const ucp_proto_init_params_t *init_params)
+{
+    ucp_context_t *context                = init_params->worker->context;
+    uint64_t rndv_modes                   = UCS_BIT(UCP_RNDV_MODE_RKEY_PTR);
+    ucp_proto_single_init_params_t params = {
+        .super.super         = *init_params,
+        .super.cfg_thresh    = ucp_proto_rndv_cfg_thresh(context, rndv_modes),
+        .super.cfg_priority  = 0,
+        .super.overhead      = 0,
+        .super.latency       = 0,
+        .super.min_length    = 0,
+        .super.max_length    = SIZE_MAX,
+        .super.min_frag_offs = UCP_PROTO_COMMON_OFFSET_INVALID,
+        .super.max_frag_offs = UCP_PROTO_COMMON_OFFSET_INVALID,
+        .super.max_iov_offs  = UCP_PROTO_COMMON_OFFSET_INVALID,
+        .super.hdr_size      = 0,
+        .super.memtype_op    = UCT_EP_OP_LAST,
+        .super.flags         = UCP_PROTO_COMMON_INIT_FLAG_RKEY_PTR |
+                               UCP_PROTO_COMMON_INIT_FLAG_RECV_ZCOPY |
+                               UCP_PROTO_COMMON_INIT_FLAG_REMOTE_ACCESS |
+                               UCP_PROTO_COMMON_INIT_FLAG_SINGLE_FRAG,
+        .lane_type           = UCP_LANE_TYPE_RKEY_PTR,
+        .tl_cap_flags        = 0,
+    };
+
+    if ((init_params->select_param->op_id != UCP_OP_ID_RNDV_RECV) ||
+        (init_params->select_param->dt_class != UCP_DATATYPE_CONTIG)) {
+        return UCS_ERR_UNSUPPORTED;
+    }
+
+    return ucp_proto_single_init(&params);
```

Comment:
check explicitly that have valid remote key in https://github.com/openucx/ucx/blob/f98ff97c7ed88bf1b7f3f3cbdee9ca28968e57e0/src/ucp/proto/proto_common.c#L314
- cannot skip for rkey_ptr proto, even if NEED_MEMH is false there